pymove/tests/preprocessing/test_filters.py

```diff
@@ -0,0 +1,567 @@
+        [1, 39.9842, 116.319321, Timestamp('2008-10-23 05:53:06'), 13.884481484192932,
+         1.0, 13.884481484192932],
+    ], columns=cols, index=[0, 1]
+    )
+    assert_frame_equal(not_speedy, expected)
+    assert move_df.len() == 5
+
+    filters.clean_gps_speed_max_radius(move_df, speed_max=15, inplace=True)
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 2
+
+
+def test_clean_trajectories_with_few_points():
+    move_df, cols = _prepare_df_default(list_data_3)
+
+    not_few_points = filters.clean_trajectories_with_few_points(
+        move_df, min_points_per_trajectory=2, label_tid='id'
+    )
+
+    expected = DataFrame(data=[
+        [39.984093, 116.319237, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984200, 116.319321, Timestamp('2008-10-23 05:54:06'), 1],
+        [39.984222, 116.319405, Timestamp('2008-10-23 05:55:16'), 1],
+        [39.984219, 116.319420, Timestamp('2008-10-23 05:56:21'), 1],
+        [39.984199, 116.319320, Timestamp('2008-10-23 05:53:06'), 2],
+        [39.974222, 116.339404, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[0, 1, 2, 3, 4, 5]
+    )
+    assert_frame_equal(not_few_points, expected)
+    assert move_df.len() == 6
+
+    filters.clean_trajectories_with_few_points(move_df, min_points_per_trajectory=3,
+                                               label_tid='id', inplace=True)
+    expected = DataFrame(data=[
+        [39.984093, 116.319237, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984200, 116.319321, Timestamp('2008-10-23 05:54:06'), 1],
+        [39.984222, 116.319405, Timestamp('2008-10-23 05:55:16'), 1],
+        [39.984219, 116.319420, Timestamp('2008-10-23 05:56:21'), 1]
+    ], columns=cols, index=[0, 1, 2, 3]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 4
+
+
+def test_clean_trajectories_short_and_few_points():
+    move_df, cols = _prepare_df_with_long_trajectory()
+
+    not_short_and_few = filters.clean_trajectories_short_and_few_points(
+        move_df, label_id='id', min_trajectory_distance=100, min_points_per_trajectory=2
+    )
+    expected = DataFrame(data=[
+        [2, 39.984199, 116.31932, Timestamp('2008-10-23 05:53:06'), nan, nan, nan],
+        [2, 39.974222, 116.339404, Timestamp('2008-10-23 05:53:11'), 2039.4197149375298,
+         5.0, 407.883942987506]
+    ], columns=cols, index=[4, 5]
+    )
+    assert_frame_equal(not_short_and_few, expected)
+    assert move_df.len() == 6
+
+    filters.clean_trajectories_short_and_few_points(
+        move_df, label_id='id', min_trajectory_distance=100, min_points_per_trajectory=3,
+        inplace=True
+    )
+    expected = DataFrame(data=[
+        [1, 39.984093, 116.319237, Timestamp('2008-10-23 05:53:05'), nan, nan, nan],
+    ], columns=cols, index=[0]
+    )
+    expected.drop(index=0, inplace=True)
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 0
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)